### ISSUE TITLE
Is it possible to check the buffer size in the FastGetSolutionValue() (with KRATOS_DEBUG)?

### ISSUE THREAD

**Issue by loumalouomega** [KratosMultiphysics/Kratos#1424]
Yesterday I almost become even more crazy trying to find why the BDF2 scheme wasn't working and it was due to the buffer size. I have been checking  and thinking and I believe that right now is not possible to check the size of the buffer size in the `FastGetValue()` method (unless the `mQueueSize` is the same as the buffer_size). Would this work? (kratos/containers/variables_list_data_value_container.h)

~~~cpp
    template<class TDataType>
    TDataType& FastGetValue(const Variable<TDataType>& rThisVariable, SizeType QueueIndex)
    {        
        #ifdef KRATOS_DEBUG       
        KRATOS_ERROR_IF((QueueIndex + 1) > mQueueSize) << "ERROR:: YOUR ARE TRYING TO ACCES A VALUE OUTSIDE THE BUFFER SIZE: " << mQueueSize << std::endl;
        #endif   
        
        return *(TDataType*)Position(rThisVariable, QueueIndex);
    }
~~~

**Comment by loumalouomega** [comment]
Related with #1421

**Comment by RiccardoRossi** [comment]
@pooyan, we need your input on this. Is QueueSize in the variables_list_data_value_container the buffer size?

**Comment by RiccardoRossi** [comment]
@loumalouomega i think that the error tag is inappropriate for this issue

**Comment by loumalouomega** [comment]
better?, debug is green, like greenlight projects

**Comment by pooyan-dadvand** [comment]
I agree with adding a check in debug. And as @RiccardoRossi mentions the `mQueueSize` is the buffer size in `VariablesListDataValueContainer`.

You can blame me for the confusion. :disappointed:

**Comment by loumalouomega** [comment]
Solved in #1435 (I am not super fast, I have it already prepared)

**Comment by loumalouomega** [comment]
Solved
